sound/soc/sof/topology.c

```diff
@@ -1590,7 +1590,7 @@ static int sof_widget_load_pga(struct snd_soc_component *scomp, int index,
 	if (!volume)
 		return -ENOMEM;
 
-	if (le32_to_cpu(tw->num_kcontrols) != 1) {
+	if (le32_to_cpu(tw->num_kcontrols) > 2) {
```

Comment:
Having more than 2 kcontrols is an error, while having 0 is not?

Reply:
hmm yeah it is me just getting it to work somehow... I'm actually now thinking can we have a pga with static volume without controls. Anyway I'm wondering do we need the check at all, so what is the min and max number of controls a widget can have?

Reply:
I'm not sure you should even have a limit, other than available resources as a matter of fact. While right now we can't think of any reason to allow more than two, who knows what the future reserves? And since you say 0 may be valid, the only check should be that the value is valid (nonnegative)... And if that field is unsigned (I don't know if it is) you may thus drop the check completely I suppose.

Reply:
volume widget without at least 1 control doesnt make sense. so check for > 0 would be more appropriate

Reply:
ok, done